dpnp/dpnp_iface_mathematical.py

```diff
@@ -1759,35 +1786,70 @@ def round_(a, decimals=0, out=None):
     return around(a, decimals, out)
 
 
-def sign(x1, **kwargs):
+def sign(
+    x,
+    /,
+    out=None,
+    *,
+    order="K",
+    where=True,
+    dtype=None,
+    subok=True,
+    **kwargs,
+):
     """
     Returns an element-wise indication of the sign of a number.
 
     For full documentation refer to :obj:`numpy.sign`.
 
+    Returns
+    -------
+    out : dpnp.ndarray
+        The indication of the sign of each element of `x`.
+
     Limitations
     -----------
-    Parameter `x1` is supported as :class:`dpnp.ndarray`.
-    Keyword arguments `kwargs` are currently unsupported.
+    Parameters `x` is only supported as either :class:`dpnp.ndarray` or :class:`dpctl.tensor.usm_ndarray`.
+    Parameters `where`, `dtype` and `subok` are supported with their default values.
+    Keyword argument `kwargs` is currently unsupported.
     Otherwise the function will be executed sequentially on CPU.
     Input array data types are limited by supported DPNP :ref:`Data types`.
 
     Examples
     --------
     >>> import dpnp as np
-    >>> result = np.sign(np.array([-5., 4.5]))
-    >>> [x for x in result]
-    [-1.0, 1.0]
-
-    """
-
-    x1_desc = dpnp.get_dpnp_descriptor(
-        x1, copy_when_strides=False, copy_when_nondefault_queue=False
-    )
-    if x1_desc and not kwargs:
-        return dpnp_sign(x1_desc).get_pyobj()
-
-    return call_origin(numpy.sign, x1, **kwargs)
+    >>> np.sign(np.array([-5., 4.5]))
+    array([-1.0, 1.0])
+    >>> np.sign(np.array(0))
+    array(0)
+    >>> np.sign(np.array(5-2j))
+    array([1+0j])
+
+    """
+
+    if numpy.iscomplexobj(x):
+        return call_origin(
```

Comment:
I believe `x / dpnp.sqrt(x * x)` would be much more faster than `call_origin` here.
In general, it looks we need a separate kernel to handle that, since the numpy result doesn't match dpct.tensor implementation. But it should be Ok to temporary workaround that through existing calls, rather than writing new kernel, which will definitely take much more time.

I would propose to extend `dpnp_sign` and to add a special handling of complex type there

Reply:
`dpctl.tensor.sign` already returns `x/|x|` and numpy returns `sign(x.real) + 0j if x.real != 0 else sign(x.imag) + 0j`. Since, we do not have `dpnp.real` and `dpnp.imag` yet, it is not straightforward to add a special handling complex type here.

Reply:
then looks we have to state in the limitations block regarding limited support of complex type